### DISCUSSION TITLE
HDF5 file: Where do I find the tracks associated with the instances?

### DISCUSSION THREAD

**Discussion by cosimogonnelli** [talmolab/sleap#944] [Help!]
Hi Team,

I exported a prediction file as .h5 using 'exporting analysis HDF5' via the GUI. 
I started looking into the file following some of the code in here: https://sleap.ai/notebooks/Analysis_examples.html
I started looking into each list of the dataset: 

['edge_inds', 'edge_names', 'instance_scores', 'labels_path', 'node_names', 'point_scores', 'provenance', 'track_names', 'track_occupancy', 'tracking_scores', 'tracks', 'video_ind', 'video_path']

I can't locate or understand how each single track is 'attached' to each single instance. I thought I was going to find some list or dict for example.  I need to figure this out since I am planning to work on some code that can reassign the right track to the right animal instance when there is a swap of tracks. 

Please let me if you need more info

Thanks as usual!

**Comment by roomrys** [comment]
Hi @cosimogonnelli,

Sounds like you are interested in the locations matrix (see the "Loading the data" section of the [Analysis examples notebook](https://colab.research.google.com/github/talmolab/sleap/blob/main/docs/notebooks/Analysis_examples.ipynb)). The shape of the locations matrix is (number of frames, number of nodes, xy coordinates, number of tracks). Thus, the last dimension of the location matrix is where instances are attached to tracks. 

If there are no animals in a track at a certain frame, then the coordinates will be `[np.nan, np.nan]` (initialized [here](https://github.com/talmolab/sleap/blob/b2cd605a2a87c59ac91058b7065ce904a9402ad3/sleap/info/write_tracking_h5.py#L119)). To switch tracks post-SLEAP, we could make changes to this locations matrix by swapping which coordinates are stored in at which track index. However, note that any corrections made to the analysis file are post-SLEAP.

Let us know if you need any help and/or have any more questions.

Thanks,
Liezl
